### PR TITLE
Add User access and User password support

### PR DESCRIPTION
**Pull request by canteuni**
This pull request aims to add support for : 
- Get/SetUserAccess commands (manage privilege and permissions for IPMI users)
- SetUserPassword command (manage user password, enable / disable IPMI user)

### CODE REVIEW

File: pyipmi/messaging.py [hunk 1]
```diff
@@ -86,3 +136,26 @@ def __str__(self):
...
         return s
+
+class UserAccess(State):
+
+    def _from_response(self, rsp):
+        self.user_count = rsp.max_user.max_user
+        self.enabled_user_count = rsp.enabled_user.count
+        self.enabled_status = rsp.enabled_user.status
+        self.fixed_name_user_count = rsp.fixed_names.count
+        self.privilege_level = CONVERT_RAW_TO_USER_PRIVILEGE.get(rsp.channel_access.privilege, UserPrivilegeLevel.RESERVED)
```
Comment (hthiery) [inline pyipmi/messaging.py:171]: please try not to exceed the line length of 79 chars

File: pyipmi/messaging.py [hunk 2]
```diff
@@ -96,6 +96,30 @@ def set_user_access(self, userid, ipmi_msg, link_auth, callback_only,
         rsp = self.send_message(req)
         check_completion_code(rsp.completion_code)
 
+    def set_user_password(self, userid, password=''):
+        if len(password) > 20:
```
Comment (hthiery) [inline pyipmi/messaging.py:100]: why 20 if you use ljust(16)?

File: pyipmi/messaging.py [hunk 3]
```diff
@@ -96,6 +96,30 @@ def set_user_access(self, userid, ipmi_msg, link_auth, callback_only,
...
+            raise ValueError("Password length cannot be greater than 20.")
+        req = create_request_by_name('SetUserPassword')
+        req.userid.userid = userid
+        req.operation.operation = 0b10
```
Comment (hthiery) [inline pyipmi/messaging.py:104]: can we have an ENUM or define here?

File: pyipmi/messaging.py (hunk 2 above)
Comment (canteuni) [reply pyipmi/messaging.py:100]: Oh you're right, my bad. I mixed up between IPMI v1.5 and v2.0

File: pyipmi/messaging.py (hunk 3 above)
Comment (canteuni) [reply pyipmi/messaging.py:104]: sure I'll do that

**Comment by hthiery** [comment]
thank you

**Pull request merged**